portfolio/src/main/java/com/google/sps/data/TimeRangeGroup.java

```diff
@@ -0,0 +1,97 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** Models a collection of Time Ranges with methods for add and delete. */
+public class TimeRangeGroup {
+
+  public List<TimeRange> allTimeRanges;
+
+  public TimeRangeGroup() {
```

Comment:
Consider adding a constructor that takes a collection of time ranges. And the constructor can perform the merging behavior mentioned above.